pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/dictionary/DoubleMutableDictionary.java

```diff
@@ -155,20 +168,67 @@ public boolean inRange(String lower, String upper, int indexOfValueToCompare, bo
 
   @Override
   public String getStringValue(int dictionaryId) {
-    return ((Double) getRawValueFromBiMap(dictionaryId)).toString();
-  }
-
-  private double getDouble(int dictionaryId) {
-    return (Double) dictionaryIdBiMap.get(dictionaryId);
+    return String.valueOf(_dictionaryIdBiMap.getKey(dictionaryId));
   }
 
   @Override
   public Object getMinVal() {
-    return min;
+    return _min;
   }
 
   @Override
   public Object getMaxVal() {
-    return max;
+    return _max;
+  }
+
+  @Override
+  public void print() {
+    System.out.println("************* printing dictionary for column : " + _column + " ***************");
+    for (double value : _dictionaryIdBiMap.keySet()) {
+      System.out.println(value + "," + _dictionaryIdBiMap.getValue(value));
+    }
+    System.out.println("************************************");
+  }
+
+  public static class Double2IntBiMap {
+    private Double2IntMap _valueToDictIdMap;
+    private DoubleArrayList _dictIdToValueMap;
+
+    public Double2IntBiMap() {
+      _valueToDictIdMap = new Double2IntOpenHashMap();
+      _dictIdToValueMap = new DoubleArrayList();
+      _valueToDictIdMap.defaultReturnValue(INVALID_ID);
+    }
+
+    public void put(double value) {
```

Comment:
return boolean to indicate if newEntry was added or not and use it to avoid updating min max